src/ring_theory/polynomial/cyclotomic/basic.lean

```diff
@@ -703,6 +742,74 @@ begin
   exact (is_primitive_root_exp n hpos.ne').is_integral hpos,
 end
 
+/-- `cyclotomic n ℚ` is irreducible. -/
+lemma cyclotomic.irreducible_rat {n : ℕ} (hpos : 0 < n) : irreducible (cyclotomic n ℚ) :=
+begin
+  rw [← map_cyclotomic_int],
+  exact (is_primitive.int.irreducible_iff_irreducible_map_cast (cyclotomic.is_primitive n ℤ)).1
+    (cyclotomic.irreducible hpos),
+end
+
+/-- If `n ≠ m`, then `(cyclotomic n ℚ)` and `(cyclotomic m ℚ)` are coprime. -/
+lemma cyclotomic.is_coprime {n m : ℕ} (h : n ≠ m) : is_coprime (cyclotomic n ℚ) (cyclotomic m ℚ) :=
+begin
+  by_cases hnzero : n = 0,
+  { rw [hnzero, cyclotomic_zero],
+    exact is_coprime_one_left },
+  by_cases hmzero : m = 0,
+  { rw [hmzero, cyclotomic_zero],
+    exact is_coprime_one_right },
+  exact (irreducible.coprime_iff_not_dvd $ cyclotomic.irreducible_rat $ zero_lt_iff.2 hnzero).2
+    (λ hdiv, h $ cyclotomic_injective $ eq_of_monic_of_associated (cyclotomic.monic n ℚ)
+    (cyclotomic.monic m ℚ) $ irreducible.associated_of_dvd (cyclotomic.irreducible_rat
+    (zero_lt_iff.2 hnzero)) (cyclotomic.irreducible_rat (zero_lt_iff.2 hmzero)) hdiv),
+end
```

Comment:
```suggestion
lemma cyclotomic.is_coprime {n m : ℕ} (h : n ≠ m) : is_coprime (cyclotomic n ℚ) (cyclotomic m ℚ) :=
begin
  rcases n.eq_zero_or_pos with rfl | hnzero,
  { exact is_coprime_one_left },
  rcases m.eq_zero_or_pos with rfl | hmzero,
  { exact is_coprime_one_right },
  rw (irreducible.coprime_iff_not_dvd $ cyclotomic.irreducible_rat $ hnzero),
  exact (λ hdiv, h $ cyclotomic_injective $ eq_of_monic_of_associated (cyclotomic.monic n ℚ)
    (cyclotomic.monic m ℚ) $ irreducible.associated_of_dvd (cyclotomic.irreducible_rat
    hnzero) (cyclotomic.irreducible_rat hmzero) hdiv),
end
```

a slight golf; also I'd probably change the name to `is_coprime_rat`; I wish we could just pass the `irreducible` arguments in but I think it becomes rapidly unwieldy.

Reply:
Well, the human proof is just "they are irreducible so it's ok", but indeed we always forget that even an irreducible has divisors... and Lean is just very precise (and of course `cyclotomic 0 R` is not irreducible). So I think this is just the proof.